src/uicomponents/uiapexcard.cpp

```diff
@@ -0,0 +1,294 @@
+      lv_snprintf(dsc->text, dsc->text_length, "%s", apexcard->values[index].first.c_str());
+      dsc->radius=90;
+    }
+    else {
+      lv_snprintf(dsc->text, dsc->text_length, "");
+    }
+  }
+}
+
+UIApexCard::UIApexCard(HABackend &_backend, const std::string _panel, int _index, lv_obj_t* _parent) :
+  UIEntity(nullptr, _parent)
+{
+
+  // auto state = entity->getJsonState();
+  // std::cerr << "INITIAL STATE FOR " << _entity->name << ":" << state.dump(2) << std::endl;
+
+  // if (!state.contains("attributes")) {
+  //   throw std::runtime_error("RGBLight can't operate with a state that has no attributes");
+  // }
+
+  // auto attributes = state["attributes"];
+  // if (!attributes.contains("supported_color_modes") || !attributes["supported_color_modes"].is_array()) {
+  //   throw std::runtime_error("RGBLight really needs to know the supported_color_modes as an array");
+  // }
+  // std::vector<std::string> supported_color_modes = attributes["supported_color_modes"].get<std::vector<string>>();
+
+  // // FIXME: we get all the supported color modes here, but we don't support all of them. Missing is RGBW, RGBWW and WHITE
+  // for (auto& mode : supported_color_modes) {
+  //   std::cerr << "    supported color mode:" << mode << std::endl;
+  //   std::transform(mode.begin(), mode.end(), mode.begin(), [](unsigned char _c) { return std::tolower(_c); }); // this needed?
+  //   if (mode == "unknown") {
+  //     throw std::runtime_error("support color mode is 'unknown', we really can't deal with that.");
+  //   }
+  //   if (mode == "brightness") {
+  //     showBrightness = true;
+  //   }
+  //   if (mode == "color_temp") { // FIXME: HA docs say "and it's color temperature is present in the state".
+  //     showBrightness = true;
+  //     showColorTemp = true;
+  //   }
+  //   if (mode == "hs") {
+  //     showColorWheel = true;
+  //     showBrightness = true;
+  //   }
+  //   if (mode == "rgbw") {
+  //     showColorWheel = true;
+  //     showBrightness = true;
+  //   }
+  //   if (mode == "rgb" || mode == "rgbww") {
+  //     showColorWheel = true;
+  //     showBrightness = true;
+  //     showColorTemp = true;
+  //   }
+  //   if (mode == "white") {
+  //     showWhite = true;
+  //   }
+  // }
+
+  // // We generate a UI based on 'supported_color_modes'. color_mode then tells us which mode to use. Color_mode should be in uiupdate()
+  flowpanel = lv_obj_create(_parent);
+  lv_obj_set_width(flowpanel, uiEntityWidth*3);
+  lv_obj_set_height(flowpanel, 450);
+  // lv_obj_set_style_pad_all(flowpanel, 5, LV_PART_MAIN | LV_STATE_DEFAULT);
+  // lv_obj_set_align(flowpanel, LV_ALIGN_CENTER);
+  // lv_obj_set_flex_flow(flowpanel, LV_FLEX_FLOW_COLUMN);
+  // lv_obj_set_flex_align(flowpanel, LV_FLEX_ALIGN_START, LV_FLEX_ALIGN_CENTER, LV_FLEX_ALIGN_START);
+
+  // lv_coord_t widthheight = uiEntityWidth - (lv_coord_t)50;
+
+  lv_obj_t* label = createLabel(flowpanel, "Grafiekje"); // FIXME somehow this is not showing
```

Comment:
i copied this from uirgblight, where it does work. However, this pointer should go into class/object storage. Noting this here so we can fix it in uirgblight too.

Reply:
@cyclops1982 mentioned something like "it is okay if only LVGL owns a pointer to this, we currently believe that LVGL will free the object when it is cleaning up the parent"